### PR TITLE
Sandbox

### PR DESCRIPTION
**Pull request by sneakyweasel**
- Created a dedicated sandbox page with the ability to edit individual cell and level informations.
- Added a dblClick on AppCell to permanently select a cell and offer editing abilities
- Integrated conditional cell form display
- Works with the current JSON export
- Still WIP, waiting for your feedback

### CODE REVIEW

**Comment by stared** [comment]
Screenshots plz! :)

**Comment by sneakyweasel** [comment]
cell-editor

**Comment by sneakyweasel** [comment]
So it allows fine tuning of every property of cells, double click then change element, drag sliders, etc.
Still WIP but getting there!

**Comment by stared** [comment]
It is certainly interesting!

**Comment by sneakyweasel** [comment]
sugar
Simple, double and triple sugar, we should see with Klem for variable graphics

File: src/components/EncyclopediaPage/EncyclopediaTransition.vue [hunk 1]
```diff
@@ -30,7 +30,10 @@
 <script lang="ts">
 import { Vue, Prop, Component } from 'vue-property-decorator'
 import { Elem } from '@/engine/interfaces'
-import { Coord, Grid, Cell } from '@/engine/classes'
+import Coord from '@/engine/Coord'
+import Cell from '@/engine/Cell'
+import Grid from '@/engine/Grid'
```
Comment (JStrebeyko) [inline src/components/EncyclopediaPage/EncyclopediaTransition.vue]: Was there something wrong with the `@/engine/classes` form of import?

File: src/components/SandboxPage/LevelEditor.vue [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  @Prop() readonly level!: Level
+  Elem = Elem
+  Group = Group
+}
+</script>
+
+<style lang="scss" scoped>
+.small {
+  font-size: 10px;
+}
```
Comment (JStrebeyko) [inline src/components/SandboxPage/LevelEditor.vue]: the CSS class is not utilized.

File: src/components/AppModal.vue [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    </div>
+  </transition>
+</template>
+
+<script lang="ts">
+import { Vue } from 'vue-property-decorator'
+export default class AppModal extends Vue {
+  // close(): void {
+  //   this.$emit('close')
+  // }
```
Comment (JStrebeyko) [inline src/components/AppModal.vue]: this is not needed, well played in template

**Comment by stared** [comment]
In the current form, it may be hard to merge due to conflicting changes.

In general, I advise creating new interfaces with a different component and make them accessible with `/dev/our-new-component` path, so that at the same time it is easy to test and share it, but also the risk of:

* backward-incompatible changes,
* conflicts with other branches.

**Comment by stared** [comment]
I did try to run it, but there are quite a few things that do not work, including the sandbox:

image

**Comment by stared** [comment]
@sneakyweasel a lot of nice ideas, but this is a code that is too far to be merged.
Could you create a post in Basecamp with these screenshots so we will discuss sandbox special parts?

**Pull request closed without merging**